### PR TITLE
Implement IpDiscoveryService

### PR DESCRIPTION
**Pull request by metonimie**
I'm gonna read tomorrow more about GATT stuff.

Right now I've managed to implement the IpDiscoveryService by looking at your examples.

Please review my commit, running ```python BLEHandler.py``` throws an error for me.

### CODE REVIEW

File: BLECarPiZeroW/BLEHandler.py [hunk 1]
```diff
@@ -62,6 +85,7 @@ class BLEAdvertisement(Advertisement):
     def __init__(self, bus, index):
         Advertisement.__init__(self, bus, index, 'peripheral')
         self.add_service_uuid(MovementService.MVMT_SVC_UUID)
+        self.add_service_uuid(IpDiscoveryService.IPDSC_SVC_UUID)
```
Comment (vnemes) [inline BLECarPiZeroW/BLEHandler.py]: Remove this line

File: BLECarPiZeroW/BLEHandler.py [hunk 2]
```diff
@@ -43,6 +44,28 @@ def WriteValue(self, value, options):
...
+
+class IpDiscoveryService(Service):
+    IPDSC_SVC_UUID = 'a23242a7-e7df-4f09-9450-071765271de1'
+
+    def __init__(self, bus, index):
+        Service.__init__(self, bus, index, self.IPDSC_SVC_UUID, True)
+        self.add_characteristic(LanIpCharacteristic(bus, 0, self))
+
 class MovementService(Service):
     MVMT_SVC_UUID = '4fafc201-1fb5-459e-8fcc-c5c9c331914b'
```
Comment (vnemes) [inline BLECarPiZeroW/BLEHandler.py]: Add the IP Service to be registered to the BLE Application aswell (@line 81):
```
class BLEApplication(Application):
    def __init__(self, bus):
        Application.__init__(self, bus)
        self.add_service(MovementService(bus, 0))
        self.add_service(IpDiscoveryService(bus, 1))
```

File: BLECarPiZeroW/BLEHandler.py [hunk 3]
```diff
@@ -62,6 +85,7 @@ class BLEAdvertisement(Advertisement):
     def __init__(self, bus, index):
         Advertisement.__init__(self, bus, index, 'peripheral')
         self.add_service_uuid(MovementService.MVMT_SVC_UUID)
+        self.add_service_uuid(IpDiscoveryService.IPDSC_SVC_UUID)
         self.include_tx_power = True
```
Comment (vnemes) [inline BLECarPiZeroW/BLEHandler.py]: Add a new advertisement for the IP Service here, to be registered in main.
```
class IPAdvertisement(Advertisement):
    def __init__(self, bus, index):
        Advertisement.__init__(self, bus, index, 'peripheral')
        self.add_service_uuid(IpDiscoveryService.IPDSC_SVC_UUID)
        self.include_tx_power = True
```
Inside main, add:
```
ip_advertisement = IPAdvertisement(bus,1)
[...]
# Register advertisement for IP Service
    ad_manager.RegisterAdvertisement(ip_advertisement.get_path(), {},
                                     reply_handler=register_ad_cb,
                                     error_handler=register_ad_error_cb)
```

**Comment by metonimie** [comment]
Right, will do this after I come home from work.

**Comment by metonimie** [comment]
Done.

**Pull request merged**